src/plopp/backends/plotly/canvas.py

```diff
@@ -0,0 +1,280 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+from ...core.utils import maybe_variable_to_number
+
+import plotly.graph_objects as go
+import scipp as sc
+from typing import Dict, Literal, Tuple, Union
+
+
+class Canvas:
+    """
+    Matplotlib-based canvas used to render 2D graphics.
+    It provides a figure and some axes, as well as functions for controlling the zoom,
+    panning, and the scale of the axes.
+
+    Parameters
+    ----------
+    ax:
+        If supplied, use these axes to create the figure. If none are supplied, the
+        canvas will create its own axes.
+    cax:
+        If supplied, use these axes for the colorbar. If none are supplied, and a
+        colorbar is required, the canvas will create its own axes.
+    figsize:
+        The width and height of the figure, in inches.
+    title:
+        The title to be placed above the figure.
+    grid:
+        Display the figure grid if ``True``.
+    vmin:
+        The minimum value for the vertical axis. If a number (without a unit) is
+        supplied, it is assumed that the unit is the same as the current vertical axis
+        unit.
+    vmax:
+        The maximum value for the vertical axis. If a number (without a unit) is
+        supplied, it is assumed that the unit is the same as the current vertical axis
+        unit.
+    aspect:
+        The aspect ratio for the axes.
+    scale:
+        Change axis scaling between ``log`` and ``linear``. For example, specify
+        ``scale={'tof': 'log'}`` if you want log-scale for the ``tof`` dimension.
+    cbar:
+        Add axes to host a colorbar if ``True``.
+    """
+
+    def __init__(self,
+                 figsize: Tuple[float, float] = None,
+                 title: str = None,
+                 grid: bool = False,
+                 vmin: Union[sc.Variable, int, float] = None,
+                 vmax: Union[sc.Variable, int, float] = None,
+                 aspect: Literal['auto', 'equal'] = 'auto',
+                 scale: Dict[str, str] = None,
+                 cbar: bool = False,
+                 **ignored):
+
+        self.fig = go.FigureWidget(
+            layout={
+                'modebar_remove': [
+                    'zoom', 'pan', 'select', 'toImage', 'zoomIn', 'zoomOut',
+                    'autoScale', 'resetScale', 'lasso2d'
+                ],
+                'margin': {
+                    'l': 0,
+                    'r': 0,
+                    't': 0 if title is None else 40,
+                    'b': 0
+                },
+                'dragmode':
+                False,
+                'width':
+                600 if figsize is None else figsize[0],
+                'height':
+                400 if figsize is None else figsize[1]
```

Comment:
Because I could not have a default argument further up the chain that would work for both Matplotlib and Plotly.
If I set the default to `(6.4, 4.8)` (in inches) in `FigLine` or `FigImage`, then this propagates down to the canvas for all backends, and plotly works in pixels, not in inches.
So my solution was to go with `None`.

An alternative is to have all these default values in a config file, but I've been holding off using a config file for now, as i find it can mess things up and we already have a config file for scipp, maybe we shouldn't have too many.